mypy/plugin.py

```diff
@@ -379,3 +421,332 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+        attributes = OrderedDict()  # type: OrderedDict[str, Attribute]
+
+        for stmt in body:
+            if isinstance(stmt, AssignmentStmt) and isinstance(stmt.lvalues[0], NameExpr):
+                lhs = stmt.lvalues[0]
+                assert isinstance(lhs, NameExpr)
+                name = lhs.name
+                typ = stmt.type
+
+                func_name = called_function(stmt.rvalue)
+
+                if func_name in attr_attrib_makers:
+                    assert isinstance(stmt.rvalue, CallExpr)
+                    # Look for default=<something> in the call.
+                    # TODO: Check for attr.NOTHING
+                    attr_has_default = bool(get_argument(stmt.rvalue, 'default'))
+
+                    # If the type isn't set through annotation but it is passed through type=
+                    # use that.
+                    type_arg = get_argument(stmt.rvalue, 'type')
+                    if type_arg and not typ:
+                        try:
+                            un_type = expr_to_unanalyzed_type(type_arg)
+                        except TypeTranslationError:
+                            ctx.api.fail('Invalid argument to type', type_arg)
+                        else:
+                            typ = ctx.api.anal_type(un_type)
+                            if typ and isinstance(lhs.node, Var) and not lhs.node.type:
+                                # If there is no annotation, add one.
+                                lhs.node.type = typ
+                                lhs.is_inferred_def = False
+
+                    # If the attrib has a converter function take the type of the first argument
+                    # as the init type.
+                    converter = get_argument(stmt.rvalue, 'converter')
+                    convert = get_argument(stmt.rvalue, 'convert')
+
+                    if convert and converter:
+                        ctx.api.fail("Can't pass both `convert` and `converter`.", stmt.rvalue)
+                    elif convert:
+                        # Note: Convert is deprecated but works the same.
+                        converter = convert
+
+                    if (converter
+                            and isinstance(converter, RefExpr)
+                            and converter.node
+                            and isinstance(converter.node, FuncBase)
+                            and converter.node.type
+                            and isinstance(converter.node.type, CallableType)
+                            and converter.node.type.arg_types):
+                        typ = converter.node.type.arg_types[0]
+
+                    # Does this even have to go in init?
+                    init = get_bool_argument(stmt.rvalue, 'init', True)
+
+                    if name in attributes:
+                        del attributes[name]
+                    attributes[name] = Attribute(name, typ, attr_has_default, init, stmt)
```

Comment:
Add a comment that we first delete and them set to get right order in ``OrderedDict`` (this is my guess).